### DISCUSSION TITLE
Bulk assign 500 items creates '414 Request-URI Too Large' error

### DISCUSSION THREAD

**Discussion by novocent** [laravel/nova-issues#5555] [Q&A]
Any suggestion on how to place an arbitrary limit on a resource? I want to be able to select 500 records and run an action on them. (Bulk assign to 500 items to a Sales Rep at a time) 

When I tried the perPageOptions in combination with select this page, a value of 500 created a 414 Request-URI Too Large error and the actions dropdown failed to load. Increasing the large_client_header_buffers in nginx can remove the error but the actions still refuse to show properly. 

I tried to make a filter, but the limit doesn’t seem to be respected. 

```namespace App\Nova\Filters;

use Laravel\Nova\Filters\Filter;
use Laravel\Nova\Http\Requests\NovaRequest;
use Log;

class RecordFilter extends Filter
{
    /**
     * The filter's component.
     *
     * @var string
     */
    public $component = 'select-filter';

    /**
     * Apply the filter to the given query.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @param  mixed  $value
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public function apply(NovaRequest $request, $query, $value)
    {                

        //Log::debug($query->toSql());

        switch ($value) {
            case 5:
                return $query->limit(5);
            case 100:
                return $query->limit(100);
            case 500:
                return $query->limit(500);
            default:
                return $query;
        }    
    }

    /**
     * Get the filter's available options.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest  $request
     * @return array
     */
    public function options(NovaRequest $request)
    {
        return [
            '5' => '5',
            '150' => '150',
            '500' => '500',
        ];    
    }
}
```

**Comment by crynobone** [comment]
Hi there,

With such a large number of selected resources (without using **Select All**, technically you are sending every selected resources IDs to the request query string and this would cause the web server to return those errors.

Unfortunately, we are unable to change the structure without removing existing features (show available actions based on selected resources etc). 

It might be better if you utilise "Select all" instead or use Standalone Actions to handle such requirement.

**Comment by novocent** [comment]
In my use case, a user filters through tens of thousands of records to achieve a search results. Of that result, I would like to assign a sales rep to groups of 500 records at a time. 

A standalone action does not pass the model to action. So that won't work because the records need to come from the filtered result.  Using a standard action and choosing "select all" has terrible performance because tens of thousands of models have to be passed only to be ignored. That is why I was trying to pursue the filter by record count, but adding the limit here won't work in Nova.

**Comment by novocent** [comment]
Is there a better way to run an action on the first X number of records in the result of a filtered resource?

**Reply by crynobone** [reply]
1. I would believe filters is still available for Standalone Actions.
2. You can add a select field to add limit and then generate the query using available classes from Nova.

**Reply by novocent** [reply]
How can you access the current query within the action?

**Reply by crynobone** [reply]
Generate a new query, standalone action doesn't provide that.
